### PR TITLE
Update runtime to 47

### PR DESCRIPTION
**Pull request by A6GibKm**
cc @alatiera.

I do not have a scanner to test this, it would be appreciated if someone can test that.

### CODE REVIEW

File: org.gnome.SimpleScan.json [hunk 1]
```diff
@@ -196,14 +196,18 @@
                 "--with-cupsbackenddir=/app/lib/cups/backend"
             ],
             "build-options": {
-                "cppflags": "-I/app/include/libusb-1.0"
+                "cppflags": "-I/app/include/libusb-1.0 -Wno-error=implicit-function-declaration -Wno-error=return-mismatch -Wno-error=incompatible-pointer-types"
```
Comment (alatiera) [inline org.gnome.SimpleScan.json:199]: This is certainly not the correct fix, there is a problem with symbols/headers being gone and needs to be debugged. This very likely explodes at runtime atm

File: org.gnome.SimpleScan.json [hunk 2]
```diff
@@ -196,14 +196,18 @@
...
             },
             "sources": [
                 {
                     "type": "archive",
                     "url": "https://downloads.sourceforge.net/sourceforge/hplip/hplip-3.23.12.tar.gz",
                     "sha256": "a76c2ac8deb31ddb5f0da31398d25ac57440928a0692dcb060a48daa718e69ed"
                 },
+                {
+                    "type": "patch",
+                    "path": "hplip-hpaio-gcc14.patch"
```
Comment (hfiguiere) [inline org.gnome.SimpleScan.json:209]: have you tried just setting `-fpermissive` through `cflags` ? This usually work around most of the issues.

Comment (A6GibKm) [reply org.gnome.SimpleScan.json:221]: Yeah, that makes things *compile*, but can also make them explode during runtime.

**Comment by A6GibKm** [comment]
I fixed the remaining issues. I would highly appreciate if anyone tests whether scanning a document actually works. This now includes https://github.com/flathub/org.gnome.SimpleScan/pull/31.

File: org.gnome.SimpleScan.json [hunk 3]
```diff
@@ -207,15 +207,39 @@
...
+                    "path": "hplip-covscan.patch"
+                },
+                {
+                    "type": "patch",
+                    "path": "hplip-pcardext-disable.patch"
+                },
+                {
+                    "type": "patch",
+                    "path": "hplip-orblite-use-const.patch"
+                },
```
Comment (hfiguiere) [inline org.gnome.SimpleScan.json:242]:
```suggestion
                {
                    "type": "patch",
                    "paths": [
                        "hplip-scan-hpaio-include.patch",
                        "hplip-hpaio-gcc14.patch"
                        "hplip-scan-orblite-c99.patch",
                        "hplip-covscan.patch",
                        "hplip-pcardext-disable.patch",
                        "hplip-orblite-use-const.patch"
                    ]
                },
```

less lines.

Comment (A6GibKm) [reply org.gnome.SimpleScan.json:242]: Since when do we have that?

Comment (hfiguiere) [reply org.gnome.SimpleScan.json:242]: I'd say forever. 

I would have to check the changelog for a more accurate answer.

Comment (hfiguiere) [reply org.gnome.SimpleScan.json:242]: patch is from August 2018. Before 1.0 according to git describe.

File: org.gnome.SimpleScan.json [hunk 4]
```diff
@@ -207,15 +207,26 @@
...
                         "project-id": 1327,
                         "stable-only": true,
                         "url-template": "https://downloads.sourceforge.net/sourceforge/hplip/hplip-$version.tar.gz"
                     }
                 },
+                {
+                    "type": "patch",
+                    "paths": [
+                        "hplip-scan-hpaio-include.patch",
+                        "hplip-hpaio-gcc14.patch"
```
Comment (hfiguiere) [inline org.gnome.SimpleScan.json:223]:
```suggestion
                        "hplip-hpaio-gcc14.patch",
```

it was missing a ','

**Comment by A6GibKm** [comment]
bot, build

**Pull request merged**